src/Microsoft.VisualStudio.ProjectSystem.Managed/Annotations.cs

```diff
@@ -85,4 +85,61 @@ internal sealed class DoesNotReturnIfAttribute : Attribute
         /// <summary>Gets the condition parameter value.</summary>
         public bool ParameterValue { get; }
     }
+
+    /// <summary>Specifies that the method or property will ensure that the listed field and property members have not-null values.</summary>
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Property, Inherited = false, AllowMultiple = true)]
+    internal sealed class MemberNotNullAttribute : Attribute
```

Comment:
For my information: how are these picked up for nullability analysis?

Reply:
The C# compiler looks for attributes with a certain name and shape. If it finds them, it can use the information they provide as part of its nullability analysis. It doesn't care if they come from source or referenced assemblies.

Reply:
These attributes are not needed when targeting .NET Core 3.0 or later, where they're included in the core framework. We're targeting .NET Framework so have to define them. The compiler will happily use them.